sql/core.go

```diff
@@ -966,6 +966,36 @@ type StoredProcedureDetails struct {
 	ModifiedAt      time.Time // The time of the last modification to the stored procedure.
 }
 
+// ExternalStoredProcedureDetails are the details of an external stored procedure. Compared to standard stored
+// procedures, external ones are considered "built-in", in that they're not created by the user, and may not be modified
+// or deleted by a user. In addition, they're implemented as a function taking standard parameters, compared to stored
+// procedures being implemented as expressions.
+type ExternalStoredProcedureDetails struct {
+	// Name is the name of the external stored procedure. If two external stored procedures share a name, then they're
+	// considered overloaded. Standard stored procedures do not support overloading.
+	Name string
+	// Schema describes the row layout of the RowIter returned from Function.
+	Schema Schema
+	// Function is the implementation of the external stored procedure. All functions should have the following definition:
+	// `func(*Context, <PARAMETERS>) (RowIter, error)`. The <PARAMETERS> may be any of the following types: `bool`,
+	// `string`, `[]byte`, `int8`-`int64`, `uint8`-`uint64`, `float32`, `float64`, `time.Time`, or `decimal`
+	// (shopspring/decimal). The architecture-dependent types `int` and `uint` (without a number) are also supported.
+	//
+	// Each parameter, by default, is an IN parameter. If the parameter type is a pointer, e.g. `*int32`, then it
+	// becomes an INOUT parameter. There is no way to set a parameter as an OUT parameter.
+	//
+	// Values are converted to their nearest type before being passed in, following the conversion rules of their
+	// related SQL types. The exceptions are `time.Time` (treated as a `DATETIME`), string (treated as a `LONGTEXT` with the
+	// default collation) and decimal (treated with a larger precision and scale). Take extra care when using decimal
+	// for an INOUT parameter, to ensure that the returned value fits the original's precision and scale, else an error
+	// will occur.
+	//
+	// As functions support overloading, each variant must have a completely unique function signature to prevent
+	// ambiguity. Uniqueness is determined by the number of parameters. If two functions are returned that have the same
```

Comment:
If we're going to support overloading (which I agree we should), then it should be an error for DB to return two procedures with the same name and number of params.

We can refine this later (like allow the same number of params but different types), but be conservative and make it an error for now.